jetty-core/jetty-security/src/main/java/org/eclipse/jetty/security/Authenticator.java

```diff
@@ -0,0 +1,137 @@
+import org.eclipse.jetty.util.Callback;
+
+/**
+ * Authenticator Interface
+ * <p>
+ * An Authenticator is responsible for checking requests and sending
+ * response challenges in order to authenticate a request.
+ * Various types of {@link Authentication} are returned in order to
+ * signal the next step in authentication.
+ *
+ * @version $Rev: 4793 $ $Date: 2009-03-19 00:00:01 +0100 (Thu, 19 Mar 2009) $
+ */
+public interface Authenticator
+{
+
+    /**
+     * Configure the Authenticator
+     *
+     * @param configuration the configuration
+     */
+    void setConfiguration(AuthConfiguration configuration);
+
+    /**
+     * @return The name of the authentication method
+     */
+    String getAuthMethod();
+
+    /**
+     * Called prior to validateRequest. The authenticator can
+     * manipulate the request to update it with information that
+     * can be inspected prior to validateRequest being called.
+     * The primary purpose of this method is to satisfy the Servlet
+     * Spec 3.1 section 13.6.3 on handling Form authentication
+     * where the http method of the original request causing authentication
+     * is not the same as the http method resulting from the redirect
+     * after authentication.
+     *
+     * @param request the request to prepare for authentication
+     */
+    default Request prepareRequest(Request request)
+    {
+        return request;
+    }
+
+    /**
+     * Validate a request
+     *
+     * @param request The request
+     * @param response The response
+     * @param callback the callback to use for writing a response
+     * @param mandatory True if authentication is mandatory.
+     * @return An Authentication.  If Authentication is successful, this will be a {@link User}. If a response has
+     * been sent by the Authenticator (which can be done for both successful and unsuccessful authentications), then the result will
+     * implement {@link Authentication.ResponseSent}.  If Authentication is not mandatory, then a
+     * {@link Authentication.Deferred} may be returned.
+     * @throws ServerAuthException if unable to validate request
+     */
+    Authentication validateRequest(Request request, Response response, Callback callback, boolean mandatory) throws ServerAuthException;
+
+    /**
+     * is response secure
+     *
+     * @param request the request
+     * @param response the response
+     * @param callback the callback to write a response
+     * @param mandatory if security is mandatory
+     * @param validatedUser the user that was validated
+     * @return true if response is secure
+     * @throws ServerAuthException if unable to test response
+     */
+    @Deprecated
+    default boolean secureResponse(Request request, Response response, Callback callback, boolean mandatory, User validatedUser) throws ServerAuthException
+    {
+        return true;
+    }
+
+    /**
+     * Authenticator Configuration
+     */
+    interface AuthConfiguration
```

Comment:
I agree it would be better, but there are naming clashes as `OpenIdConfiguration` is already used, so I can't rename `OpenIdAuthConfiguration`.   So will leave it for now.